src/target/ch32f1.c

```diff
@@ -152,9 +155,36 @@ static bool ch32f1_flash_lock(target *t)
 	// FLASH_CR_FLOCK_CH32 bit does not exists on *regular* clones and defaults to '0' (see PM0075 for STM32F1xx)
 	if (!(cr & FLASH_CR_FLOCK_CH32)) {
```

Comment:
Same suggestion as above for how to do this return

Reply:
As this code looks like it'll work even if this does introduce a clang-tidy lint warning, we will accept and merge this as-is but fix it up on the target clean-up branch.